File: Linked List/Java/GFG - FIND LENGTH OF LOOP.java
// GFG - Find length of Loop
// QUESTION LINK - https://www.geeksforgeeks.org/problems/find-length-of-loop/1

class Solution {
    int countNodesinLoop(Node head) {
        Node slow = head;
        Node fast = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;

            if (slow == fast) {
                // Loop detected, count the length
                int count = 1;
                slow = slow.next;
                while (slow != fast) {
                    count++;
                    slow = slow.next;
                }
                return count;
            }
        }

        return 0;
    }
}

File: Linked List/Python/GFG - FIND LENGTH OF LOOP.py
# GFG - Find length of Loop
# QUESTION LINK - https://www.geeksforgeeks.org/problems/find-length-of-loop/1

class Solution:
    def countNodesinLoop(self, head):
        slow = head
        fast = head

        while fast is not None and fast.next is not None:
            slow = slow.next
            fast = fast.next.next

            if slow == fast:
                # Loop detected, count the length
                count = 1
                slow = slow.next
                while slow != fast:
                    count += 1
                    slow = slow.next
                return count

        return 0
